packages/fast-usdc/src/exos/liquidity-pool.js

```diff
@@ -89,60 +102,145 @@ export const prepareLiquidityPoolKit = (zone, zcf, USDC, tools) => {
      */
     (shareMint, node) => {
       const { brand: PoolShares } = shareMint.getIssuerRecord();
-      const proposalShapes = makeProposalShapes({ USDC, PoolShares });
-      const shareWorth = makeParity(makeDust(USDC), PoolShares);
+      const proposalShapes = makeProposalShapes({
+        USDC: usdc.brand,
+        PoolShares,
+      });
+      const shareWorth = makeParity(makeDust(usdc.brand), PoolShares);
       const { zcfSeat: poolSeat } = zcf.makeEmptySeatKit();
-      const shareWorthRecorderKit = tools.makeRecorderKit(node, RatioShape);
+      const poolMetricsRecorderKit = tools.makeRecorderKit(
+        node,
+        PoolMetricsShape,
+      );
+      /** used for `checkPoolBalance` invariant */
+      const outstandingLends = makeEmpty(usdc.brand);
+      const poolStats = /** @type {PoolStats} */ harden({
+        totalFees: makeEmpty(usdc.brand),
+        totalBorrows: makeEmpty(usdc.brand),
+        totalReturns: makeEmpty(usdc.brand),
+      });
       return {
-        shareMint,
-        shareWorth,
+        outstandingLends,
+        poolStats,
+        poolMetricsRecorderKit,
         poolSeat,
         PoolShares,
         proposalShapes,
-        shareWorthRecorderKit,
+        shareMint,
+        shareWorth,
       };
     },
     {
-      feeSink: {
+      assetManager: {
+        lookupBalance() {
+          const { poolSeat } = this.state;
+          return poolSeat.getAmountAllocated('USDC', usdc.brand);
+        },
         /**
          * @param {Amount<'nat'>} amount
-         * @param {Payment<'nat'>} payment
+         * @returns {Promise<Payment<'nat'>>}
          */
-        async receive(amount, payment) {
-          const { poolSeat, shareWorth } = this.state;
-          const { external } = this.facets;
-          await depositToSeat(
-            zcf,
-            poolSeat,
-            harden({ USDC: amount }),
-            harden({ USDC: payment }),
+        async borrow(amount) {
+          const { outstandingLends, poolSeat, poolStats } = this.state;
+          const available = poolSeat.getAmountAllocated('USDC', usdc.brand);
+          // Validate amount is available in pool
+          const post = borrowCalc(
+            amount,
+            available,
+            outstandingLends,
+            poolStats,
+          );
+
+          // XXX COMMIT POINT?
+          const { USDC: paymentP } = await withdrawFromSeat(zcf, poolSeat, {
+            USDC: amount,
+          });
```

Comment:
as discussed, rather than dealing with payments here, let's have the caller supply a seat.
The advancer can make a temp seat.

Then I think we can avoid turn boundaries in `borrow` and `repay`.